src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/convert_gather_compression.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/cpu_opset/common/op/gather_compression.hpp"
+#include "convert_gather_compression.hpp"
+#include "openvino/op/matmul.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include <transformations/utils/utils.hpp>
+#include "transformations/rt_info/keep_const_precision.hpp"
+
+#include "openvino/opsets/opset10.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+#include "itt.hpp"
+
+ov::intel_cpu::ConvertToGatherCompression::ConvertToGatherCompression() {
+    MATCHER_SCOPE(ConvertToGatherCompression);
+    /**
+     * Dequantization subgraph pattern
+     *
+     * Input
+     *    \
+     *  Convert    zero point
+     *      \      /
+     *      Subtract    scale
+     *         \       /
+     *          Multiply
+     *             |
+     *          Convert
+     *             |
+     *           Gather
+     */
+    const element::TypeVector precisions = {ov::element::u8, ov::element::u4};
+
+    auto input_pattern = ov::pass::pattern::wrap_type<opset10::Constant>();
+    auto convert1_pattern = ov::pass::pattern::wrap_type<opset10::Convert>({input_pattern}, ov::pass::pattern::consumers_count(1));
+    auto zero_point_pattern = ov::pass::pattern::any_input();
+    auto subtract_pattern = ov::pass::pattern::wrap_type<opset10::Subtract>({convert1_pattern, zero_point_pattern});
+    auto multiply_pattern = ov::pass::pattern::wrap_type<opset10::Multiply>({subtract_pattern, ov::pass::pattern::any_input()});
+    auto multiply_no_subtract_pattern = ov::pass::pattern::wrap_type<opset10::Multiply>({convert1_pattern, ov::pass::pattern::any_input()});
+
+    auto convert2_pattern = ov::pass::pattern::wrap_type<opset10::Convert>({multiply_pattern}, ov::pass::pattern::consumers_count(1));
+    auto const_pattern = ov::pass::pattern::wrap_type<opset10::Constant>();
+    auto gather_pattern = ov::pass::pattern::wrap_type<opset8::Gather>({convert2_pattern, ov::pass::pattern::any_input(), const_pattern});
```

Comment:
from `GatherCompression::execReferenceU8/U4` we can see so far this new CPU node only supports :
 - 2D data&indices with axis==0;
 - dequantization sub-graph is per-row;

So we must add more checks in callback or builds explicit pattern to ensure these 2 conditions

Reply:
Done.